syntax = "proto3";
package federation.fed_protobuff;
import "federation/fed_protobuff/collaborative_learning.proto";

service EvalServing {

    // Send a model for evaluation.
    rpc ModelEvaluationOnFederationValidationSets(stream ModelEvaluationRequest) returns (DoubleValue) {}

    // Unary RPC. One time request for retrieving execution metadata from evaluator.
    rpc RetrieveValidationMetadataFromEvaluator (Learner) returns (JsonStringValue) {}

}

// Extend LearnerExecutionResult with an additional attribute to indicate community model or not.
message ModelEvaluationRequest {
    oneof inherit_learner_execution_result { LearnerExecutionResult learner_execution_result = 1; }
    bool is_community_model = 2;
}